Makefile

```diff
@@ -194,6 +194,12 @@ setup-pre-commit:
 		cp ./scripts/pre-commit .git/hooks/pre-commit; \
 		chmod +x .git/hooks/pre-commit;	\
 	fi
+	@if [ ! -f .git/hooks/commit-msg ]; then \
+		echo "Installing commit-msg hook for DCO validation..."; \
+		echo '#!/bin/bash' > .git/hooks/commit-msg; \
+		echo 'bash scripts/check-dco.sh "$$1"' >> .git/hooks/commit-msg; \
+		chmod +x .git/hooks/commit-msg; \
```

Comment:
wondering if we can move this to `pre-push` or have `prepare-commit-msg`. Consider a scenario where someone just wants to add a commit just to save that piece of code but doesn't wants to push that commit.